metapool/sample_sheet.py

```diff
@@ -1044,10 +1046,27 @@ def make_sample_sheet(metadata, table, sequencer, lanes, strict=True):
         sheet._add_metadata_to_sheet(metadata, sequencer)
         sheet._add_data_to_sheet(table, sequencer, lanes, metadata['Assay'],
                                  strict)
-        return sheet
-    else:
-        for message in messages:
-            message.echo()
+
+        # now that we have a SampleSheet() object, validate it for any
+        # additional errors that may have been present in the data and/or
+        # metadata.
+        messages = sheet.quiet_validate_and_scrub_sample_sheet()
+
+        if not any([isinstance(m, ErrorMessage) for m in messages]):
```

Comment:
just to confirm, is it possible to get non ErrorMessage's ?

Reply:
It is possible to get WarningMessages. If there are only WarningMessages and no ErrorMessages, the SampleSheet will be returned to the user. In this case, the user won't see the WarningMessages since they are echoed below.